deps/ncrypto/ncrypto.cc

```diff
@@ -220,4 +221,75 @@ int PasswordCallback(char* buf, int size, int rwflag, void* u) {
   return -1;
 }
 
+// ============================================================================
+// SPKAC
+
+bool VerifySpkac(const char* input, size_t length) {
+#ifdef OPENSSL_IS_BORINGSSL
+  // OpenSSL uses EVP_DecodeBlock, which explicitly removes trailing characters,
+  // while BoringSSL uses EVP_DecodedLength and EVP_DecodeBase64, which do not.
+  // As such, we trim those characters here for compatibility.
+  length = std::string(input.data()).find_last_not_of(" \n\r\t") + 1;
+#endif
+  NetscapeSPKIPointer spki(
+      NETSCAPE_SPKI_b64_decode(input, length));
+  if (!spki)
+    return false;
+
+  EVPKeyPointer pkey(X509_PUBKEY_get(spki->spkac->pubkey));
+  return pkey ? NETSCAPE_SPKI_verify(spki.get(), pkey.get()) > 0 : false;
+}
+
+BIOPointer ExportPublicKey(const char* input, size_t length) {
+  BIOPointer bio(BIO_new(BIO_s_mem()));
+  if (!bio) return BIOPointer();
+
+#ifdef OPENSSL_IS_BORINGSSL
+  // OpenSSL uses EVP_DecodeBlock, which explicitly removes trailing characters,
+  // while BoringSSL uses EVP_DecodedLength and EVP_DecodeBase64, which do not.
+  // As such, we trim those characters here for compatibility.
+  length = std::string(input).find_last_not_of(" \n\r\t") + 1;
+#endif
+  NetscapeSPKIPointer spki(
+      NETSCAPE_SPKI_b64_decode(input, length));
+  if (!spki) return BIOPointer();
+
+  EVPKeyPointer pkey(NETSCAPE_SPKI_get_pubkey(spki.get()));
+  if (!pkey) return BIOPointer();
+
+  if (PEM_write_bio_PUBKEY(bio.get(), pkey.get()) <= 0) return BIOPointer();
+
+  return std::move(bio);
+}
+
+Buffer<char> ExportChallenge(const char* input, size_t length) {
+#ifdef OPENSSL_IS_BORINGSSL
+  // OpenSSL uses EVP_DecodeBlock, which explicitly removes trailing characters,
+  // while BoringSSL uses EVP_DecodedLength and EVP_DecodeBase64, which do not.
+  // As such, we trim those characters here for compatibility.
+  length = std::string(input).find_last_not_of(" \n\r\t") + 1;
```

Comment:
```suggestion
  length = std::string(input, length).find_last_not_of(" \n\r\t") + 1;
```

We can make this a string view to reduce 1 string allocation